### DISCUSSION TITLE
Setting the root path for model download

### DISCUSSION THREAD

**Discussion by hmf** [KoljaB/RealtimeSTT#155] [Q&A] [answered]
Hello,

I have experimented in Linux with faster-whisper and would like to try RealtimSTT (using the git repo code). I have very little disc space and have therefore downloaded the models to an alternate path (not the default `.cache` directory) using faster-whisper. I see that we have a `model_path` that is assigned to faster whisper's `WhisperModel.model_size_or_path`. What I would like is to set is `WhisperModel.download_root`. 

I could change [this](https://github.com/KoljaB/RealtimeSTT/blob/67879852d564fdac4318557840ebb7420bf52bf3/RealtimeSTT/audio_recorder.py#L133) and hard-code the root. But before I go that route, I would like to know if there is a "better" to go about this.

TIA

**Comment by KoljaB** [comment, marked as answer]
Good point! RealtimeSTT doesn't expose a parameter for download_root from faster-whisper currently. Adding that could be useful, but for now, hard-coding it is probably your best option. If you’re up for it, feel free to open a PR, or let me know if I should look into it for a future update.

**Reply by hmf** [reply]
Create PR https://github.com/KoljaB/RealtimeSTT/pull/158. 

Here are some tests I performed:

```shell
python -m tests.realtimestt_test --root /mnt/ssd2/hmf/VSCodeProjects/faster-whisper/models
python -m tests.realtimestt_test -m tiny -d /mnt/ssd2/hmf/VSCodeProjects/faster-whisper/models
python -m tests.realtimestt_test -m tiny -r tiny.en -d /mnt/ssd2/hmf/VSCodeProjects/faster-whisper/models
python -m tests.realtimestt_test -l en -m tiny -r tiny.en -d /mnt/ssd2/hmf/VSCodeProjects/faster-whisper/models

python -m tests.realtimestt_test -m tiny --root /mnt/ssd2/hmf/VSCodeProjects/faster-whisper/models
python -m tests.realtimestt_test --model tiny --root /mnt/ssd2/hmf/VSCodeProjects/faster-whisper/models
python -m tests.realtimestt_test --model tiny --rt-model tiny.en --root /mnt/ssd2/hmf/VSCodeProjects/faster-whisper/models
python -m tests.realtimestt_test --lang en --model tiny --rt-model tiny.en --root /mnt/ssd2/hmf/VSCodeProjects/faster-whisper/models


python -m tests.realtimestt_test --model distil-small.en --root /mnt/ssd2/hmf/VSCodeProjects/faster-whisper/models
```
